index.html

```diff
@@ -722,47 +1502,116 @@ <h2>
       </p>
       <p>
-        A <a>user agent</a> MUST NOT dispatch this event type if the
-        <a>environment settings object</a> is a <a>non-secure context</a>.
+        A [=user agent=] MUST NOT dispatch this event type if the [=environment
+        settings object=] is a [=non-secure context=].
       </p>
     </section>
     <section>
       <h3 id="event-gamepaddisconnected">
         The <dfn class="event">gamepaddisconnected</dfn> event
       </h3>
+      <p>
+        When a gamepad becomes unavailable on the system, run the following
+        steps:
+      </p>
+      <ol>
+        <li>If the [=current settings object=]'s [=environment settings object
+        / responsible document=] is not [=allowed to use=] the `"gamepad"`
+        permission, then abort these steps.
+        </li>
+        <li>Let |gamepad:Gamepad| be the {{Gamepad}} representing the
+        unavailable device.
+        </li>
+        <li>Set |gamepad|.{{Gamepad/[[connected]]}} to `false`.
+        </li>
+        <li>Let |navigator:Navigator| be |gamepad|'s [=relevant global
+        object=]'s {{Navigator}} object.
+        </li>
+        <li>If |navigator|.{{Navigator/[[hasGamepadGesture]]}} is `true` and
+        |gamepad|.{{Gamepad/[[exposed]]}} is `true`:
+          <ol>
+            <li>If |gamepad|'s [=relevant global object=] is a {{Window}}
+            object, then [=queue a global task=] on the [=gamepad task source=]
+            to [=fire an event=] named {{gamepaddisconnected}} at |gamepad|'s
+            [=relevant global object=] using {{GamepadEvent}} with its
+            {{GamepadEvent/gamepad}} attribute initialized to |gamepad|.
+            </li>
+          </ol>
+        </li>
+        <li>Set
+        |navigator|.{{Navigator/[[gamepads]]}}[|gamepad|.{{Gamepad/index}}] to
+        `null`.
+        </li>
+        <li>While |navigator|.{{Navigator/[[gamepads]]}} [=list/is not empty=]
+        and the last [=list/item=] of |navigator|.{{Navigator/[[gamepads]]}} is
+        `null`, [=list/remove=] the last [=list/item=] of
+        |navigator|.{{Navigator/[[gamepads]]}}.
+        </li>
+      </ol>
       <p>
         User agents implementing this specification must provide a new DOM
-        event, named <code>gamepaddisconnected</code>. The corresponding event
-        MUST be of type <code>GamepadEvent</code> and, if <a>allowed to use</a>
-        the "`gamepad`" permission, MUST fire on the <code>window</code>
-        object. Registration for and firing of the
-        <code>gamepaddisconnected</code> event MUST follow the usual behavior
-        of DOM Events. [[DOM]]
+        event, named {{gamepaddisconnected}}. The corresponding event MUST be
+        of type {{GamepadEvent}} and, if [=allowed to use=] the `"gamepad"`
+        permission, MUST fire on the {{Window}} object. Registration for and
+        firing of the {{gamepaddisconnected}} event MUST follow the usual
+        behavior of DOM Events. [[DOM]]
```

Comment:
Don't need this :) We are doing the right thing... 

```suggestion
        permission, MUST fire on the {{Window}} object.
```

Reply:
Removed